### ISSUE TITLE
Dreamcast Build

### ISSUE THREAD

**Issue by cjcliffe** [cjcliffe/CubicVR#1]
Need to port over the Dreamcast makefiles, test updates and reconfigure them to the new repository layout.

**Comment by cjcliffe** [comment]
DC Makefiles are now in, and example project compiles but no idea yet if the result works.

**Comment by cjcliffe** [comment]
Current DC progress can be found in the "dreamcast-experimental" branch, build instructions have been added to the wiki

**Comment by mamad25** [comment]
Hi, i have some compilation issue for CubicVr from dreamcast experimental. It's about GLX/gl.h include not found. Here's the dump of the error:

make[1]: Leaving directory `/opt/toolchains/cubicvr/CubicVR/dreamcast/libkglx'
/opt/toolchains/dc/sh-elf/bin/sh-elf-gcc -O2 -fomit-frame-pointer -ml -m4-single-only -ffunction-sections -fdata-sections -I/opt/toolchains/kos-ports/include -I/opt/toolchains/KallistiOS/include -I/opt/toolchains/KallistiOS/kernel/arch/dreamcast/include -I/opt/toolchains/KallistiOS/addons/include -I/opt/toolchains/kos-ports/include -D_arch_dreamcast -D_arch_sub_pristine -Wall -g -fno-builtin -fno-strict-aliasing -DARCH_DC -O3 -funroll-loops -D_arch_dreamcast -I../cubicvr/include/ -I../cubicvr/include/math/ -Ibuild/include/ -I./ -c -o  Resource.o ../cubicvr/source/Resource.cpp
In file included from ../cubicvr/include/CubicVR/XYZ.h:29:0,
                 from ../cubicvr/include/CubicVR/datatree/DataTree.h:37,
                 from ../cubicvr/include/CubicVR/Resource.h:27,
                 from ../cubicvr/source/Resource.cpp:28:
../cubicvr/include/CubicVR/GLExt.h:105:21: fatal error: GLX/gl.h: No such file or directory
compilation terminated.
make: **\* [Resource.o] Error 1

Thank you, i'm looking forward to use this engine for my next project!!!

**Comment by pmlopes** [comment]
It looks you're missing some header files in your system or their paths added to the includes check if you have the file "gl.h"

**Comment by mamad25** [comment]
Yes it's what i have checked in first time, but, from my dc envirronnment (i've followed the cubicvr guide in the wiki) i can compile kgl test (from nehe) and i've found in addons directory of Kalisstios libkglx.a. 
I'm going to check if there another library to install for having GLX/gl.h

**Comment by m-mac** [comment]
I experienced this issue as well. I managed to solve it by creating the folders GLX and GL in the dreamcast/ directory and adding the required header files in there.
